### PR TITLE
Ran provider (update) generator.

### PR DESCRIPTION
**Pull request by juliancheal**
This keeps the provider inline with new updates for all providers.
Also removed rubocop_base.yml as no longer required.

### CODE REVIEW

File: manageiq-providers-lenovo.gemspec [hunk 1]
```diff
@@ -7,11 +7,12 @@ Gem::Specification.new do |s|
...
   s.homepage    = "https://github.com/ManageIQ/manageiq-providers-lenovo"
-  s.summary     = "Lenovo XClarity Provider for ManageIQ"
-  s.description = "Lenovo XClarity Provider for ManageIQ"
+  s.summary     = "Lenovo Provider for ManageIQ"
+  s.description = "Lenovo Provider for ManageIQ"
   s.licenses    = ["Apache-2.0"]
 
   s.files = Dir["{app,config.lib}/**/*"]
 
-  s.add_dependency "xclarity_client", "~> 0.3.1"
```
Comment (bdunne) [inline manageiq-providers-lenovo.gemspec]: Looks like this was a mistaken deletion

File: .travis.yml [hunk 2]
```diff
@@ -10,15 +10,5 @@ env:
...
-after_script: bundle exec codeclimate-test-reporter
-notifications:
-  webhooks:
-    urls:
-    - https://webhooks.gitter.im/e/994cc93494715ae5990f
-    on_success: change
-    on_failure: always
-    on_start: never
+install: bin/setup
+after_script: bundle exec codeclimate-test-reporter
```
Comment (bdunne) [inline .travis.yml]: Missing trailing newline.

File: manageiq-providers-lenovo.gemspec (hunk 1 above)
Comment (juliancheal) [inline manageiq-providers-lenovo.gemspec]: Good spot. Darn that @durandom's generator :)

File: .travis.yml (hunk 2 above)
Comment (juliancheal) [inline .travis.yml]: Ah yeah, I think @durandom mentioned that it's missing in the generator, so I'll add it there too.

File: .rubocop_cc.yml [hunk 3]
```diff
@@ -1,4 +1,5 @@
 inherit_from:
 # this is downloaded by .codeclimate.yml
-- .rubocop_cc_mangeiq.yml
```
Comment (Fryguy) [inline .rubocop_cc.yml]: I just noticed the old code said "mangeiq" 😆

Comment (juliancheal) [inline .rubocop_cc.yml]: @Fryguy I need some sort of spell check in my IDE I think.

File: .travis.yml [hunk 4]
```diff
@@ -10,15 +10,5 @@ env:
   - RUBY_GC_HEAP_GROWTH_FACTOR=1.25
 addons:
   postgresql: '9.4'
-before_install:
-- source ${TRAVIS_BUILD_DIR}/tools/ci/before_install.sh
-script:
-- bundle exec bin/rails app:test:providers:lenovo:setup app:test:providers:lenovo
-after_script: bundle exec codeclimate-test-reporter
-notifications:
```
Comment (durandom) [inline .travis.yml]: the notifications should stay

Comment (juliancheal) [inline .travis.yml]: Done.

**Pull request merged**